paddle/fluid/operators/sequence_softmax_op.cc

```diff
@@ -29,6 +29,31 @@ class SequenceSoftmaxOp : public framework::OperatorWithKernel {
     ctx->SetOutputDim("Out", ctx->GetInputDim("X"));
     ctx->ShareLoD("X", /*->*/ "Out");
   }
+
+ protected:
+  framework::OpKernelType GetExpectedKernelType(
+      const framework::ExecutionContext& ctx) const override {
+    bool use_cudnn = ctx.Attr<bool>("use_cudnn");
+    use_cudnn &= platform::is_gpu_place(ctx.GetPlace());
+#ifdef PADDLE_WITH_CUDA
+    if (platform::is_gpu_place(ctx.GetPlace())) {
+      auto& dev_ctx =
+          ctx.template device_context<platform::CUDADeviceContext>();
+      use_cudnn &= dev_ctx.cudnn_handle() != nullptr;
+    }
+#endif
```

Comment:
The logic of getting `use_cudnn` here is complex. Could you refine it and make it more readable?

Reply:
Done.